spec.bs

```diff
@@ -1260,8 +1248,35 @@ an [=ordered map=] whose [=map/keys=] are [=strings=] and whose [=map/values=] a
     1. Set |global|'s
       [=InterestGroupBiddingScriptRunnerGlobalScope/is component auction=] to |isComponentAuction|.
     1. Set |global|'s [=InterestGroupBiddingScriptRunnerGlobalScope/interest group=] to |ig|.
-    1. Return the result of [=evaluating a script=] with |global|, |script|, "`generateBid`",
+    1. Let |startTime| be the current time.
+    1. Let |result| be the result of [=evaluating a script=] with |global|, |script|, "`generateBid`",
       |arguments|, and |timeout|.
+    1. Let |duration| be the current time minus |startTime| in milliseconds.
+    1. If |global|'s [=InterestGroupBiddingScriptRunnerGlobalScope/priority=] is not null:
+      1. Set |ig|'s [=interest group/priority=] to |global|'s
+        [=InterestGroupBiddingScriptRunnerGlobalScope/priority=].
+      1. Store |ig| in the browser’s [=interest group set=].
+    1. If |global|'s [=InterestGroupBiddingScriptRunnerGlobalScope/priority signals=] is not
+      [=map/is empty|empty=]:
+      1. [=map/For each=] |k| → |v| of |global|'s
+        [=InterestGroupBiddingScriptRunnerGlobalScope/priority signals=]:
+        1. If |v| is null, [=map/remove=] |ig|'s [=interest group/priority signals overrides=][|k|].
+        1. Otherwise, [=map/set=] |ig|'s [=interest group/priority signals overrides=][|k|] to |v|.
+      1. Store |ig| in the browser’s [=interest group set=].
+    1. If |result| is a [=throw completion=] with \[[Value]] of null, then return |global|'s
+      [=InterestGroupBiddingScriptRunnerGlobalScope/bid=].

```

Comment:
If the bidding code actually executes `throw null;`, do you want this branch to be taken? If not, it would probably be better to have 'evaluate a script' return something else, probably something other than a completion record like "timeout", to signal a timeout. It's fine if you want to do that in a separate PR.

Reply:
Actually our implementation uses the setBid() value whenever the script doesn't return normally: https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:content/services/auction_worklet/bidder_worklet.cc;l=1227;drc=d9095892389e9b1be3b9068feb1330d87f35c61e;bpv=1;bpt=1
So I simplified the spec to match that properly.